community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogEntryVersion.java

```diff
@@ -126,13 +129,24 @@ public CommandReader newCommandReader()
         {
             return new PhysicalLogNeoCommandReaderV2_2_4();
         }
+    },
+    // as of 2016-05-24: neo4j 2.2.10 legacy index IndexDefineCommand maps write size as short instead of byte
+    // -8 is picked, -5 and -6 can be found in the future (2.3 and 3.0)
+    // log entry layout hasn't changed since 2_2_4 so just use that one
+    V2_2_10( -8, LogEntryParsersV2_2_4.class )
+    {
+        @Override
+        public CommandReader newCommandReader()
+        {
+            return new PhysicalLogCommandReaderV2_2_10();
+        }
     };
 
-    public static final LogEntryVersion CURRENT = V2_2_4;
+    public static final LogEntryVersion CURRENT = V2_2_10;
     public static final byte NO_PARTICULAR_LOG_HEADER_FORMAT_VERSION = -1;
     private static final LogEntryVersion[] ALL = values();
-    private static final LogEntryVersion[] NEGATIVE = new LogEntryVersion[ALL.length+1]; // pessimistic size
-    private static final LogEntryVersion[] POSITIVE = new LogEntryVersion[ALL.length]; // pessimistic size
+    private static final LogEntryVersion[] NEGATIVE = new LogEntryVersion[ALL.length+5]; // pessimistic size
```

Comment:
It's simply that the array should be at least big enough to cover all versions, positive and negative, where the absolute version value is the index into that array.